tools/interop-node/feeder/feeder.go

```diff
@@ -74,16 +38,258 @@ func (feeder *BaseFeeder) FetchNewRoundInfo(ctx context.Context) {
+				return err
+			}
+
+			voteData = append(voteData, &oracletypes.VoteData{
+				Topic: oracletypes.OralceTopic_BLOCK,
+				Data:  blockDataStr,
+			})
+
+		case oracletypes.OralceTopic_OWNERSHIP:
+			nftDataStr, err := feeder.gatherNftOwnerDataString(od.Sources, uint64(round.Timestamp))
+			if err != nil {
+				feeder.abstainRoundId = round.Id
+				return err
+			}
+			voteData = append(voteData, &oracletypes.VoteData{
+				Topic: oracletypes.OralceTopic_OWNERSHIP,
+				Data:  nftDataStr,
+			})
+		}
+	}
+
+	salt, err := GenerateSalt()
+	if err != nil {
+		return fmt.Errorf("failed to generate salt: %w", err)
+	}
+
+	if err := feeder.sendPrevote(ctx, voteData, salt, round.Id); err != nil {
+		return fmt.Errorf("failed to send prevote: %w", err)
+	}
+
+	feeder.lastPreVote = &BlockVoteInfo{
+		RoundId:  round.Id,
+		Salt:     salt,
+		VoteData: voteData,
+	}
+
+	return nil
+}
+
+// HandleAbstain Handles a prevote period when block data string cannot be gathered
+func (feeder *Feeder) HandleAbstain(ctx context.Context) error {
+	round := feeder.currentRound
+	if feeder.lastPreVote != nil && round.Id == feeder.lastPreVote.RoundId {
+		feeder.logger.Debug("already prevote, skipping this period...")
+		return nil
+	}
+
+	feeder.logger.Info("failed to gather block data string, abstaining from this vote period")
+
+	salt, err := GenerateSalt()
+	if err != nil {
+		return fmt.Errorf("failed to generate salt: %w", err)
+	}
+
+	// abstain from voting
+	abstainData := []*oracletypes.VoteData{}
+	if err := feeder.sendPrevote(ctx, abstainData, salt, round.Id); err != nil {
+		return fmt.Errorf("failed to send abstain prevote: %w", err)
+	}
+
+	feeder.lastPreVote = &BlockVoteInfo{
+		RoundId:  round.Id,
+		Salt:     salt,
+		VoteData: abstainData,
+	}
+
+	return nil
+}
+
+func (feeder *Feeder) gatherBlockDataString(chainIds []string, timestamp uint64) ([]string, error) {
+	blockDataList := make([]oracletypes.BlockData, len(chainIds))
+	for idx, cid := range chainIds {
+		bd, err := feeder.getOldestBlock(cid, timestamp)
+		if err != nil {
+			return nil, err
+		}
+
+		blockDataList[idx] = bd
+	}
+
+	return BlockDataListToBlockDataString(blockDataList), nil
+}
+
+func (feeder *Feeder) getOldestBlock(chainId string, timestamp uint64) (bd oracletypes.BlockData, err error) {
+	cc, ok := feeder.subscribers[chainId]
+	if !ok {
+		return bd, fmt.Errorf("chain client not found for chain id: %s", chainId)
+	}
+
+	return cc.GetOldestBlock(timestamp)
+}
+
+func (feeder *Feeder) gatherNftOwnerDataString(nftIds []string, timestamp uint64) ([]string, error) {
+	s := make([]string, len(nftIds))
+	for i, nftId := range nftIds {
+		nft, err := oracletypes.ParseNftId(nftId)
+		if err != nil {
+			return nil, err
+		}
+
+		bd, err := feeder.getOldestBlock(nft.ChainId, timestamp)
+		if err != nil {
+			return nil, err
+		}
+
+		cc, ok := feeder.subscribers[nft.ChainId]
+		if !ok {
+			return nil, fmt.Errorf("chain client not found for chain id: %s", nft.ChainId)
```

Comment:
We should not prematurely return here. Even if one of the `nft.ChainId` is wrong, we still have to iterate and find owners of other NFTs. 
```suggestion
			feeder.logger.Error("chain client not found for chain id: %s", nft.ChainId)
			continue
```

Reply:
> We should not prematurely return here. Even if one of the `nft.ChainId` is wrong, we still have to iterate and find owners of other NFTs.

It makes sense. However, in the current implementation of the Oracle Module, if the feeder does not submit proof of NFT ownership for a specific blockchain, it is marked as a miss for that round. Thus, sending an abstain vote might be a better option for the feeder. We could also consider changing the implementation of the Oracle Module to a more flexible approach that allows for partial proof submissions in the future.